tests/unit/plugins/test_publisher.py

```diff
@@ -16,21 +16,476 @@
+    event_msg_data['resourceRecords'] = [resource_record]
+    event_message.data = event_msg_data
+
+    error = ("Issue connecting to www.googleapis.com:"
+             " 404, message='Not Found'")
+
+    resource_record_changes = {
+        'name': 'service.nurit.com.',
+        'rrdatas': ['127.10.20.2'],
+        'type': 'A',
+        'ttl': 3600,
+        'kind': 'dns#resourceRecordSet'
+    }
+
+    changes = {'kind': 'dns#change',
+               'deletions': [resource_record_changes]}
+    publisher_instance.http_client._request_post_mock.side_effect = \
+        exceptions.GCPHTTPError(error)
+
+    await publisher_instance.publish_changes(event_message)
+
+    actual_msg = caplog.records[1].msg
+
+    expected_msg = ('[msg-1234]: DROPPING: Fatal exception '
+                    'occurred when handling message: '
+                    f'Error: {error} for changes: {changes}.')
+
+    assert expected_msg == actual_msg
+
+
+@pytest.mark.asyncio
+async def test_failed_adding_bad_rrdata(publisher_instance,
+                                        event_message,
+                                        caplog, event_msg_data_bad_rrdata):
+    """Test error is raised when trying to add record with bad rrdata"""
+    event_message.data = event_msg_data_bad_rrdata
+
+    error = "Issue connecting to www.googleapis.com: 400, message='Bad Request'"
+
+    changes = {
+        'kind': 'dns#change',
+        'additions':
+            [{'name': 'service.nurit.com.',
+              'rrdatas': ['127.10.20.899'],
+              'type': 'A',
+              'ttl': 3600,
+              'kind': 'dns#resourceRecordSet'
+              }]
+    }
+
+    publisher_instance.http_client._request_post_mock.side_effect = \
+        exceptions.GCPHTTPError(error)
+
+    await publisher_instance.publish_changes(event_message)
+
+    actual_msg = caplog.records[1].msg
+
+    expected_msg = ('[msg-1234]: DROPPING: Fatal exception '
+                    'occurred when handling message: '
+                    f'Error: {error} for changes: {changes}.')
+
+    assert expected_msg == actual_msg
+
+    msg = await publisher_instance.error_channel.get()
+    assert msg == event_message
+
+
+@pytest.mark.asyncio
+async def test_failed_unexpected_error_google(publisher_instance,
+                                              event_message,
+                                              caplog):
+    """Error is raised when getting unexpected error from google API"""
+
+    error = ("Issue connecting to www.googleapis.com: 500, "
+             "message='Backend Error'")
+
+    publisher_instance.http_client._request_post_mock.side_effect = \
+        exceptions.GCPHTTPError(error)
+
+    await publisher_instance.publish_changes(event_message)
+
+    actual_msg = caplog.records[1].msg
+
+    expected_msg = ('[msg-1234]: RETRYING: Exception occurred '
+                    f'when handling message: {error}.')
+
+    assert expected_msg == actual_msg
+
```

Comment:
This is purely a style thing, but it's confusing to me why there are so many blank lines inside these test methods, which makes them a little harder to read since there's no logical grouping to the code.  I usually group into three sections: set up (mocks, etc), performing the action to test, comparing results, and then only more subsections as needed for readability.

Reply:
ok, so I changed to set up, apply the main func, another setup to extract the log msg and then check the results.. tell me what do you think...